src/main/java/com/mongodb/kafka/connect/sink/StartedMongoSinkTask.java

```diff
@@ -103,22 +110,11 @@ private void bulkWriteBatch(final List<MongoProcessedSinkRecordData> batch) {
               .bulkWrite(writeModels, BULK_WRITE_OPTIONS);
       LOGGER.debug("Mongodb bulk write result: {}", result);
       checkRateLimit(config);
-    } catch (MongoException e) {
-      LOGGER.warn(
-          "Writing {} document(s) into collection [{}] failed.",
-          writeModels.size(),
-          namespace.getFullName());
-      handleMongoException(config, writeModels, e);
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
       throw new DataException("Rate limiting was interrupted", e);
-    } catch (Exception e) {
-      if (config.logErrors()) {
-        LOGGER.error("Failed to write mongodb documents", e);
-      }
-      if (!config.tolerateErrors()) {
-        throw new DataException("Failed to write mongodb documents", e);
-      }
+    } catch (RuntimeException e) {
+      handleTolerableWriteException(batch, e, config.logErrors(), config.tolerateErrors());
```

Comment:
Nit: Its seems a little odd to use batch here but unpack it straight away in the method to a list of SinkRecords but unpack the configs here.